backend/internal/system/immutable_resource/composite_helpers.go

```diff
@@ -0,0 +1,438 @@
+) error {
+	id := getID(entity)
+
+	// Check if it's an immutable resource
+	exists, err := fileExists(id)
+	if err != nil {
+		return err
+	}
+	if exists {
+		// Resource exists in file store - immutable
+		return immutableError
+	}
+
+	// Resource not found in file store - safe to update in database
+	return dbUpdate(entity)
+}
+
+// CompositeDeleteHelper blocks deletion of immutable resources.
+//
+// Parameters:
+//   - id: The resource ID to delete
+//   - fileExists: Function to check if ID exists in file store
+//   - dbDelete: Function to delete from database store
+//   - immutableError: Error to return when trying to delete immutable resource
+//
+// Returns:
+//   - Error if resource is immutable or deletion fails
+//
+// Behavior:
+//  1. Check if exists in file store
+//  2. If immutable, return immutableError
+//  3. If error occurs during check, return that error
+//  4. If not found, proceed with deletion
+func CompositeDeleteHelper(
+	id string,
+	fileExists func(string) (bool, error),
+	dbDelete func(string) error,
+	immutableError error,
+) error {
+	// Check if it's an immutable resource
+	exists, err := fileExists(id)
+	if err != nil {
+		return err
+	}
+	if exists {
+		// Resource exists in file store - immutable
+		return immutableError
+	}
+
+	// Resource not found in file store - safe to delete from database
+	return dbDelete(id)
+}
+
+// CompositeExistsHelper checks existence across both stores.
+//
+// Parameters:
+//   - id: The resource ID to check
+//   - dbExists: Function to check existence in database store
+//   - fileExists: Function to check existence in file store
+//
+// Returns:
+//   - true if exists in either store, false otherwise
+//   - Error if check fails
+//
+// Behavior:
+//  1. Check database store first
+//  2. If exists, return true
+//  3. Otherwise, check file store
+//  4. Return true if found in file store
+func CompositeExistsHelper(
+	id string,
+	dbExists func(string) (bool, error),
+	fileExists func(string) (bool, error),
+) (bool, error) {
+	// Check database store first
+	exists, err := dbExists(id)
+	if err != nil {
+		return false, err
+	}
+	if exists {
+		return true, nil
+	}
+
+	// Check file store
+	return fileExists(id)
+}
+
+// CompositeConflictHelper checks conflicts across both stores.
+//
+// Generic helper for any conflict check (name, handle, etc.)
+//
+// Parameters:
+//   - dbChecker: Function to check conflict in database store
+//   - fileChecker: Function to check conflict in file store
+//
+// Returns:
+//   - true if conflict exists in either store, false otherwise
+//   - Error if check fails
+//
+// Behavior:
+//  1. Check database store for conflict
+//  2. If conflict found, return true
+//  3. Otherwise, check file store
+//  4. Return true if conflict found in file store
+func CompositeConflictHelper(
+	dbChecker func() (bool, error),
+	fileChecker func() (bool, error),
+) (bool, error) {
+	// Check database store
+	dbConflict, err := dbChecker()
```

Comment:
shouldn't we check the fileExists first then dbExists ?

Reply:
Can't we refactor and have one function for this? looks like CompositeExistsHelper and CompositeConflictHelper performs the same action

Reply:
Fixed